### PR TITLE
Input can be byte or text string

### PR DESCRIPTION
**Pull request by msrb**
data-importer still runs on Python 2, therefore text string is actually
unicode, not str.

### CODE REVIEW

**Comment by msrb** [comment]
Any of you guys, PTAL. Thanks :wink:

**Comment by msrb** [comment]
Let me merge, I will test this in staging. Feel free to review, I will address all comments later.

File: test/test_graph_populator.py [hunk 1]
```diff
@@ -1,3 +1,16 @@
...
+from graph_populator import GraphPopulator
+import pytest
+
+
+def test_sanitize_text_for_query():
+    """Test GraphPopulator._sanitize_text_for_query()."""
+    f = GraphPopulator._sanitize_text_for_query
+    assert 'pkg', f('pkg')
+    assert 'desc', f(' desc\n')
+    assert 'ASL\n"2.0"', f('ASL 2.0')
```
Comment (tuxdna) [inline test/test_graph_populator.py]: This test looks like it is flipped.

Comment (msrb) [reply test/test_graph_populator.py]: Ah, very good catch. Let me see...

Comment (msrb) [reply test/test_graph_populator.py]: Ops, actually all of them are wrong. Thanks @tuxdna. PR incoming.

File: test/test_graph_populator.py [hunk 2]
```diff
@@ -1,3 +1,16 @@
...
+    assert 'ok', f('[ok]')
+    assert 'ok', f("'ok'")
+    with pytest.raises(ValueError):
+        f(100)
```
Comment (tuxdna) [inline test/test_graph_populator.py]: Proposed change:

```python
"""Tests for the graph_populator module."""

from graph_populator import GraphPopulator
import pytest


def test_sanitize_text_for_query():
    """Test GraphPopulator._sanitize_text_for_query()."""
    f = GraphPopulator._sanitize_text_for_query
    print("Hello:", f(' desc\n'))
    assert 'pkg' == f('pkg')
    assert 'desc' == f('desc\n')
    assert 'desc' == f(' desc')
    assert 'desc' == f(' desc\n')
    assert 'ASL 2.0' == f('ASL\n"2.0"')
    assert '[ok]' == f('["ok\']')
    assert 'ok' == f("'ok'")
    with pytest.raises(ValueError):
        f(100)

if __name__ == "__main__":
    test_sanitize_text_for_query()
```

File: src/graph_populator.py [hunk 3]
```diff
@@ -14,14 +15,15 @@ class GraphPopulator(object):
...
-            return ''
+        if not isinstance(text, string_types):
+            raise ValueError(
+                'Invalid query text: expected string, got {t}'.format(t=type(text))
+            )
+        # remove newlines, quotes and backslash character
+        text = " ".join([l.strip() for l in text.split("\n")])
+        text = re.sub("""['"]""", "", text)
+        text = text.replace('\\', "")
+        return text
```
Comment (tuxdna) [inline src/graph_populator.py]: Propose this change

```python
return text.strip()
```

Comment (msrb) [reply src/graph_populator.py]: Hmm, we call `l.strip()` already on the line 23, that should suffice (?)

Comment (tuxdna) [reply src/graph_populator.py]: There is a corner case when text has trailing newline:

```
In [2]: "dep\n".split("\n")
Out[2]: ['dep', '']

In [3]: " ".join("dep\n".split("\n"))
Out[3]: 'dep '
```

So either `text = text.strip()` at start of function or `return text.strip()`

**Comment by tuxdna** [comment]
Thanks for adding test cases, which caught a corner case here.

File: src/graph_populator.py (hunk 3 above)
Comment (msrb) [reply src/graph_populator.py]: I see now. Let me correct this.

**Comment by msrb** [comment]
Thanks for the review, here's new PR: https://github.com/fabric8-analytics/fabric8-analytics-data-model/pull/115
